### PR TITLE
Fix Race Condition, Add Support for Extract by Array of Sample Names (ie from a Sample Set)

### PR DESCRIPTION
**Pull request by kcibul**

### CODE REVIEW

File: scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl [hunk 1]
```diff
@@ -34,10 +35,22 @@ workflow GvsExtractCohortFromSampleNames {
     File? gatk_override
   }
 
+  # writing the array to a file has to be done in a task
+  # https://support.terra.bio/hc/en-us/community/posts/360071465631-write-lines-write-map-write-tsv-write-json-fail-when-run-in-a-workflow-rather-than-in-a-task
+  if (defined(cohort_sample_names_array)) {
+    call write_array_task {
+      input:
+        input_array = select_first([cohort_sample_names_array]),
+        docker = "gcr.io/google.com/cloudsdktool/cloud-sdk:305.0.0"
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl:44]: Pinning cloud SDK versions can help avoid bugs but this particular one is ~2 years old

File: scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl [hunk 2]
```diff
@@ -34,10 +35,22 @@ workflow GvsExtractCohortFromSampleNames {
     File? gatk_override
   }
 
+  # writing the array to a file has to be done in a task
+  # https://support.terra.bio/hc/en-us/community/posts/360071465631-write-lines-write-map-write-tsv-write-json-fail-when-run-in-a-workflow-rather-than-in-a-task
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl:39]: uh okay, TIL 🙂

File: scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl [hunk 3]
```diff
@@ -8,7 +8,8 @@ import "GvsExtractCallset.wdl" as GvsExtractCallset
 workflow GvsExtractCohortFromSampleNames {
 
   input {
-    File cohort_sample_names
+    File? cohort_sample_names
+    Array[String]? cohort_sample_names_array
```
Comment (mcovarr) [inline scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl:12]: maybe a comment here that `cohort_sample_names` will be ignored if `cohort_sample_names_array` is defined?

**Review by mcovarr — APPROVED** [review]

File: scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl (hunk 1 above)
Comment (kcibul) [reply scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl:44]: Yep - I'm just taking the same version we use in other parts of the pipelines.  We may be old but at least were consistent 😆

File: scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl (hunk 2 above)
Comment (kcibul) [reply scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl:39]: yeah -- def a bummer, but seems to be the way it is

File: scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl (hunk 3 above)
Comment (rsasch) [reply scripts/variantstore/wdl/GvsExtractCohortFromSampleNames.wdl:12]: +1

**Pull request merged**